kbs/src/api/src/attestation/mod.rs

```diff
@@ -32,6 +32,10 @@ pub trait Attest: Send + Sync {
     /// Verify Attestation Evidence
     /// Return Attestation Results Token
     async fn verify(&self, tee: Tee, nonce: &str, attestation: &str) -> Result<String>;
+
+    /// generate the challenge payload to pass to attester based on Tee and nonce
+    async fn generate_supplemental_challenge(&self, tee: Tee, tee_parameters: Option<Vec<u8>>)
```

Comment:
As `extra_params` are defined as `String`, we can directly use `tee_parameters: Option<String>` here. helps to avoid `as_bytes` and `from_utf8` on both side.

Reply:
Good point! fixed.